### PR TITLE
Title is not linked when external action

### LINKED ISSUE #948: [Teaser] - Title is not linked if first call-to-action is an external URL
## Bug Report

**Current Behavior**
If the first call-to-action is an external link, Title is not linked. If the Link field is used instead of Call-to-action it works as expected.

**Expected behavior/code**
Title needs to be linked irrespective of whether the link is internal or external when using call-to-action.

**Environment**
- AEM 6.5.2
- Core Components version 2.8

**Possible Solution**
TeaserImpl.java lines 174-175. Defect could be in Utils.java

### PR DESCRIPTION
**Pull request by dymitrs**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Fixes #948
| Patch: Bug Fix?          | Yes
| Minor: New Feature?      | No
| Major: Breaking Change?  | No
| Tests Added + Pass?      | No
| Documentation Provided   | No
| Any Dependency Changes?  | No
| License                  | Apache License, Version 2.0

Title is now visible when the first call-to-action is an external link.

### CODE REVIEW

File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/TeaserImpl.java [hunk 1]
```diff
@@ -140,6 +141,9 @@ private void initModel() {
             if (titleFromPage) {
                 if (targetPage != null) {
                     title = StringUtils.defaultIfEmpty(targetPage.getPageTitle(), targetPage.getTitle());
+                } else if (actionsEnabled) {
```
Comment (vladbailescu) [inline bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/TeaserImpl.java:144]: I think we need to also to check `!actions.isEmpty()` here

**Comment by dymitrs** [comment]
Thanks for the review. Of course, I will try to provide the missing tests & fix the existing ones that failed.

**Comment by dymitrs** [comment]
Hey, I added the missing tests. Let me know what you think about them.

**Comment by bpauli** [comment]
@dymitrs, Thank you for your contribution!

**Pull request merged**